bundles/org.openhab.binding.groupepsa/src/main/java/org/openhab/binding/groupepsa/internal/things/GroupePSAHandler.java

```diff
@@ -0,0 +1,487 @@
+            final Integer pollingIntervalM = currentConfig.getPollingInterval();
+            final Integer onlineIntervalM = currentConfig.getOnlineInterval();
+
+            if (id == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/conf-error-no-vehicle-id");
+            } else if (pollingIntervalM != null && pollingIntervalM < 1) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/conf-error-invalid-polling-interval");
+            } else if (onlineIntervalM != null && onlineIntervalM < 1) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/conf-error-invalid-online-interval");
+            } else {
+                this.id = id;
+                this.onlineIntervalM = onlineIntervalM != null ? onlineIntervalM : DEFAULT_ONLINE_INTERVAL_M;
+                startGroupePSAPolling(pollingIntervalM);
+            }
+
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+        }
+    }
+
+    @Nullable
+    public GroupePSABridgeHandler getBridgeHandler() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof GroupePSABridgeHandler) {
+                return (GroupePSABridgeHandler) handler;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void dispose() {
+        stopGroupePSAPolling();
+        id = null;
+    }
+
+    private void startGroupePSAPolling(@Nullable Integer pollingIntervalM) {
+        if (groupepsaPollingJob == null) {
+            final long pollingIntervalToUse = pollingIntervalM == null ? DEFAULT_POLLING_INTERVAL_M : pollingIntervalM;
+            groupepsaPollingJob = scheduler.scheduleWithFixedDelay(() -> pollStatus(), 1, pollingIntervalToUse * 60,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopGroupePSAPolling() {
+        ScheduledFuture<?> job = groupepsaPollingJob;
+        if (job != null) {
+            job.cancel(true);
+            groupepsaPollingJob = null;
+        }
+    }
+
+    private boolean isValidResult(VehicleStatus vehicle) {
+        return vehicle.getUpdatedAt() != null;
+    }
+
+    private boolean isConnected(VehicleStatus vehicle) {
+        Position lastPosition = vehicle.getLastPosition();
+        if (lastPosition == null) {
+            return false;
+        }
+
+        ZonedDateTime updatedAt = vehicle.getUpdatedAt();
+        if (updatedAt == null) {
+            return false;
+        }
+
+        return updatedAt.isAfter(ZonedDateTime.now().minusMinutes(onlineIntervalM));
+    }
+
+    private synchronized void updateGroupePSAState() {
+        if (System.nanoTime() - lastQueryTimeMs <= maxQueryFrequencyNanos) {
```

Comment:
Did you take a look at `ExpiringCache`? https://next.openhab.org/javadoc/latest/org/openhab/core/cache/expiringcache

Reply:
Thanks for the pointer. I think it would become a bit artificial to use it: I don't need to cache the actual value, but I want to protect it being retrieved too often (more then once a minute).